### DISCUSSION TITLE
Add Buckets

### DISCUSSION THREAD

**Discussion by Nyholmen** [scfmod/FS22_TerraFarm#73] [General]
Is there ant tutorial on how to add new buckets to this mod?

**Comment by JamieMidd** [comment]
I would like the see the “Bucket packs” From farm sim mod hub added as i like to use smaller excavators.

**Comment by gordoIsMyName** [comment]
i agree. a tutorial would be great.

not sure if you can model an implementation from the xml_configurations folder or not. i saw someone had added a small backhoe and on quick inspection it looked like they used points from the i3d config to map points and then the xml would "bind" the actions for the specified machineType.

for reference, here is the mentioned xml...

<configuration machineType="bucket" modName="Fs22_InternationalHarvester340UBackhoe" typeName="shovel" xmlName="bucket">
<terraform>
<i3d parent="0>2|2" z="0.0">
<node x="-0.998"/>
<node x="-0.998"/>
<node x="0.998"/>
<node x="0.998"/>
</i3d>
<modes>
<mode>NORMAL</mode>
<mode>LOWER</mode>
<mode>FLATTEN</mode>
<mode>SMOOTH</mode>
</modes>
</terraform>
<paint>
<i3d parent="0>2|2" z="0.0">
<node/>
</i3d>
</paint>
<discharge>
<modes>
<mode>NORMAL</mode>
<mode>RAISE</mode>
<mode>FLATTEN</mode>
<mode>SMOOTH</mode>
<mode>PAINT</mode>
</modes>
</discharge>
</configuration>
